packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts

```diff
@@ -82,26 +83,25 @@ export class RemoteTableService {
       workspaceId,
     );
 
-    if (currentRemoteTables.length === 0 || !shouldFetchPendingSchemaUpdates) {
-      const distantTablesWithStatus = Object.keys(distantTables).map(
-        (tableName) => ({
+    const shouldGetTablesWithPendingSchemaUpdates =
+      currentRemoteTables.length > 0 && shouldFetchPendingSchemaUpdates;
+
+    const distantTablesWithStatus = shouldGetTablesWithPendingSchemaUpdates
```

Comment:
I feel:
- remoteTableSchemaUpdateService should only return updates
- this should be extracted to a function that would return tables with and without updates. it feels weird to either call a service or either construct an object on the fly